### PR TITLE
Update to go with new graphql-server based on mine 5.1.0.2

### PR DESCRIPTION
**Pull request by sammyjava**
This is the update that I implemented to go with the mine-based graphql-server. The main thing I did was to replace keyword searches (which can be unreliable and insufficient) with path query searches in graphql-server, which required some minor tweaks here, plus the addition of a couple new components: `lis-publication-search-element`, `lis-qtl-search-element`, along with updates to `lis-gene-search-element` and `lis-trait-search-element`.

(Note that jekyll-legumeinfo will only contain `lis-gene-search-element` and `lis-publication-search-element` if it queries against LegumeMine since LegumeMine does not contain genetic data.)

Oh, I added some how-to style documentation as well in `create-web-component.md`.

### CODE REVIEW

File: create-web-component.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+`$ npm run build` to compile the Typescript
+
+`$ npm run serve` to run a local server for testing, which will serve the above `index.html` on port 8000.
+
+## Package your web component ##
+If your web component is working correctly,
+
+`$ npm run package` to bundle it into `dist/web-components.bundled.js`.
+
+You may now copy `web-components.bundled.js` to a web site where you want to implement your new web component.
```
Comment (alancleary) [inline create-web-component.md:316]: This is an excellent tutorial. Thanks for writing it!

**Review by alancleary — CHANGES_REQUESTED** [review]
Every thing here looks good. Please push a commit that adds the publication search and qtl search elements to the documentation. See [Static Site](https://github.com/legumeinfo/web-components#static-site) in the readme for details.

**Comment by sammyjava** [comment]
> Every thing here looks good. Please push a commit that adds the publication search and qtl search elements to the documentation. See [Static Site](https://github.com/legumeinfo/web-components#static-site) in the readme for details.

Done. That is so cool.

**Review by alancleary — APPROVED** [review]
All changes have been reviewed and tested. Approved!

**Pull request merged**